assets/js/admin-settings/rule.jsx

```diff
@@ -0,0 +1,130 @@
+/* global wp, wpLocalizedData */
+import {
+  Button,
+  SelectControl,
+  TextControl,
+  ToggleControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import React, { useEffect, useState } from 'react';
+import { useSelect } from '@wordpress/data';
+
+
+const Rule = ({
```

Comment:
Recommended props here:

- `field` - string
- `taxonomy` - string
- `term_id` - number
- `value` - string|bool (if registering the schema for this in WP gives you trouble, just make it a string, and we can translate the string `'true'` to the boolean `true`)

Additionally, add the following props as callables when certain actions take place:

- `onDelete` - func
- `onUpdate` - func

The `onUpdate` function will pass in the new values for each of the properties this control is responsible for (see above).

The parent component should be responsible for making these callback functions really simple for this component. For example:

`<Rule onDelete={() => deleteRule(index)} />`

That way, the parent component is responsible for knowing the index and this component is only responsible for displaying the current values (via props) and notifying when they update (via callbacks). Updated values will flow back down into this component via props.